src/components/rangeslider/defaults.js

```diff
@@ -38,6 +44,32 @@ module.exports = function handleDefaults(layoutIn, layoutOut, axName) {
     coerce('autorange', !axOut.isValidRange(containerIn.range));
     coerce('range');
 
+    var subplots = layoutOut._subplots;
+    if(subplots) {
+        var yIds = subplots.cartesian
+            .filter(function(subplotId) {
+                return subplotId.substr(0, subplotId.indexOf('y')) === axisIds.name2id(axName);
+            })
+            .map(function(subplotId) {
+                return subplotId.substr(subplotId.indexOf('y'), subplotId.length);
+            });
+        var yNames = Lib.simpleMap(yIds, axisIds.id2name);
+        for(var i = 0; i < yNames.length; i++) {
+            var yName = yNames[i];
+
+            var rangeContainerIn = containerIn[yName] || {};
+            var rangeContainerOut = containerOut[yName] = {};
+
+            if(rangeContainerIn.range && layoutOut[yName].isValidRange(rangeContainerIn.range)) {
+                coerceRange(rangeContainerIn, rangeContainerOut, 'rangemode', 'fixed');
+            } else {
+                coerceRange(rangeContainerIn, rangeContainerOut, 'rangemode');
+            }
+
+            coerceRange(rangeContainerIn, rangeContainerOut, 'range', layoutOut[yName].range.slice());
```

Comment:
We should only coerce `xaxis?.rangeslider.yaxis?.range` when `rangemode: 'fixed'`. For `'auto'`, the range value in punched in during `Axes.doAutoRange`. For `'match'`, the range value isn't used.

If it helps you out, I would write this block as:

```js
var rangemodeDflt;
if(rangeContainerIn.range && layoutOut[yName].isValidRange(rangeContainerIn.range)) {
     rangemodeDflt = 'fixed';
}

var rangeMode = coerceRange(rangeContainerIn, rangeContainerOut, 'rangemode', rangemodeDflt);
if(rangeMode === 'fixed') {
     coerceRange(rangeContainerIn, rangeContainerOut, 'range');
}
```

Reply:
Well, I have an issue with that :
- the rangeslider `drawis s called twice on the first render
- the second call have the wrong `axisOpts` object, the range in `rangeslider.yaxis` is not initialized by the `doAutoRange` method (which is called only once before the first `draw` call)
- it then obviously crashes when trying to access range members on this second render call

Since I've no idea where this second draw call comes from (not from my side, I've check and relayout are done before the second draw call), I've limited coerce to `rangeMode !== 'match'`.

Reply:
@TomDemulierChevret I'll take a look at this, and try to push a few last additions like a new test image. I think the double-draw is the same thing I was fighting with last night https://github.com/plotly/plotly.js/pull/2451#discussion_r172742441

Reply:
I would be glad if you could add the test image, I've some issue currently with my docker config which prevent me from doing it on my work station.

Reply:
Yep, both @etpinard and I are working on parts of this, trying to get it into v1.35 which we're hoping to release this afternoon.